platform-api/che-core-api-runner/src/main/java/org/eclipse/che/api/runner/RunnerService.java

```diff
@@ -149,55 +167,82 @@ public ApplicationProcessDescriptor getStatus(@ApiParam(value = "Workspace ID",
-                        // NotFoundException is possible and should not be treated as error in this case. Typically it occurs if slave
-                        // runner already cleaned up the task by its internal cleaner but RunQueue doesn't re-check yet slave runner and
-                        // doesn't have actual info about state of slave runner.
-                    } catch (RunnerException e) {
-                        // Decide ignore such error to be able show maximum available info.
-                        LOG.error(e.getMessage(), e);
-                    }
-                }
+        if (accountId != null) {
+            Optional<Member> owner = accountDao.getMembers(accountId)
+                                               .stream()
+                                               .filter(member -> userId.equals(member.getUserId()) &&
+                                                                 member.getRoles().contains("account/owner"))
+                                               .findAny();
+            if (!owner.isPresent()) {
+                throw new ForbiddenException("You are not an owner of the specified account");
+            }
+        }
+
+        for (RunQueueTask task : runQueue.getTasks()) {
+            // skip task if it does not run in given workspace
+            if (workspaceId != null && !task.getRequest().getWorkspace().equals(workspaceId)) {
+                continue;
+            }
+
+            // skip task if it does not run in given project
+            if (project != null && !task.getRequest().getProject().equals(project)) {
+                continue;
+            }
+
+            // skip task if it is not running by given account or by given user if account is not specified
+            if (!isRelatedToAccountOrUser(task, accountId, userId)) {
+                continue;
+            }
+
+            try {
+                processes.add(task.getDescriptor());
+            } catch (NotFoundException ignored) {
+                // NotFoundException is possible and should not be treated as error in this case. Typically it occurs if slave
+                // runner already cleaned up the task by its internal cleaner but RunQueue doesn't re-check yet slave runner and
+                // doesn't have actual info about state of slave runner.
+            } catch (RunnerException e) {
+                // Decide ignore such error to be able show maximum available info.
+                LOG.error(e.getMessage(), e);
             }
         }
         return processes;
     }
 
+    private boolean isRelatedToAccountOrUser(RunQueueTask task, String accountId, String userId) throws ServerException {
+        if (accountId != null) {
+            try {
+                return accountId.equals(workspaceDao.getById(task.getRequest().getWorkspace()).getAccountId());
+            } catch (NotFoundException e) {
+                //Workspace does not exist
```

Comment:
'Workspace does not exist' is not informative. You should write something about 'How we respond on this situation'